osu.Game/Rulesets/UI/Playfield.cs

```diff
@@ -113,6 +114,16 @@ private void load()
             }
         }
 
+        private void onNewDrawableHitObject(DrawableHitObject d)
+        {
+            d.OnNestedDrawableCreated += onNewDrawableHitObject;
+
+            OnNewDrawableHitObject(d);
+
+            Debug.Assert(!d.HasInitialized);
```

Comment:
shouldn't this assert be at the start of the method to avoid double-subscribing to `OnNestedDrawableCreated`?

Reply:
If the assertion triggers that is a bug of the code (not meant to be a recoverable error) and such a double subscription shouldn't be matter. Then, I think it is good to group the check and the set together to eliminate thoughts like what if `HasInitialized` is set in `OnNewDrawableHitObject` call above.

Reply:
I don't think anybody other than `Playfield` should really ever be mutating `IsInitialised`, though. That's another reason it makes more sense to me to guard at the start but I guess it doesn't matter much.